velox/type/tests/ConversionsTest.cpp

```diff
@@ -44,15 +44,13 @@ class ConversionsTest : public testing::Test {
     auto cast = [&](TFrom input) -> TTo {
       Expected<TTo> result;
       if (truncate & legacyCast) {
-        result = Converter<toTypeKind, void, TruncateLegacyCastPolicy>::tryCast(
-            input);
+        VELOX_NYI("No associated cast policy for truncate and legacy cast.");
```

Comment:
Why did we have TruncateLegacyCastPolicy before? Was it never used by anyone?

Reply:
I havent been able to find a use for it internally ; it also seems to pass xstream tests; nevertheless I will ask xstream folks and double check before landing.